### PR TITLE
Fix UDP integration test.

### PR DESCRIPTION
**Pull request by bemasc**
The test now passes on both Firefox and Chrome.

This resolves https://github.com/freedomjs/freedom/issues/155

### CODE REVIEW

File: spec/providers/coreIntegration/udpsocket.integration.src.js [hunk 1]
```diff
@@ -8,71 +8,70 @@ module.exports = function(provider, setup) {
   });
 
   it("Connects, has state, and sends/receives data", function (done) {
-    // Currently a copy of 'receives data' unit test from firefox, change it
+    const todo = [];  // Pending tasks to complete before calling |done()|.
```
Comment (willscott) [inline spec/providers/coreIntegration/udpsocket.integration.src.js]: We prefer ```var``` over ```const```.

Comment (bemasc) [inline spec/providers/coreIntegration/udpsocket.integration.src.js]: As you may have noticed, this test used `const` exclusively.  Would you like me to change them all?

Comment (willscott) [inline spec/providers/coreIntegration/udpsocket.integration.src.js]: I'd prefer the consistency of keeping everything as ```var```, and it
shouldn't break anything.
It's a feature where I still don't feel like I fully understand the
implications of calling soemthing const, and I'd like to see it enter
common parlance a bit more before adopting it.

Comment (bemasc) [inline spec/providers/coreIntegration/udpsocket.integration.src.js]: Done.

File: spec/providers/coreIntegration/udpsocket.integration.src.js [hunk 2]
```diff
@@ -8,71 +8,70 @@ module.exports = function(provider, setup) {
...
+    var did = function(task) {
+      var i = todo.indexOf(task);
+      expect(i).not.toEqual(-1);  // A task must not be done twice.
+      todo.splice(i, 1);  // Remove |task| from the list.
+      if (todo.length == 0) {
+        done();  // This is the only call to |done()|.
+      }
+    };
+    // callPoint returns a function that must be called exactly once.
+    var callPoint = function(name) {
```
Comment (willscott) [inline spec/providers/coreIntegration/udpsocket.integration.src.js]: maybe rename to ```requiredCallback``` or similar. I don't think ```callPoint``` describes this well.

**Comment by willscott** [comment]
One more naming thing, otherwise :+1:  - thanks for getting the ```setTimeout```s out of there!

**Comment by ryscheng** [comment]
Agreed! Thanks for taking care of this. Superbular

**Pull request merged**